Add get native window interface

xiaowei-guan opened the pull request:
We implement video player plugin with hole solution, it need engine provides get native window interface. we have to return to two types of windows(Elm window, Ecore_Wl2_window) for current status.
I think a good soluton would be to return only one type of window(Elm window), and I am trying to create elm window as root window and get ecore_wl2_window from elm window(https://github.com/flutter-tizen/engine/issues/167)

xiaowei-guan committed:
Add get ecore window interface.

xiaowei-guan committed:
Change get window method name

xiaowei-guan committed:
Fix build error

xiaowei-guan committed:
Fix build error

xiaowei-guan committed:
Add comment for getting window interface

xiaowei-guan committed:
Remove not used code

swift-kim commented on shell/platform/tizen/public/flutter_tizen.h:
```diff
@@ -17,6 +17,10 @@
 extern "C" {
 #endif
 
+// Opaque reference to a Flutter window.
+struct FlutterDesktopView;
+typedef struct FlutterDesktopView* FlutterDesktopViewRef;
```
TizenRenderer = View = Window = RenderTarget?

swift-kim commented on shell/platform/tizen/public/flutter_tizen.h:
```diff
@@ -91,6 +95,18 @@ FlutterDesktopGetPluginRegistrar(FlutterDesktopEngineRef engine,
+FLUTTER_EXPORT void* FlutterDesktopViewGetWindow(FlutterDesktopViewRef view);
+
+// Returns the view associated with this registrar's engine instance.
+FLUTTER_EXPORT FlutterDesktopViewRef FlutterDesktopPluginRegistrarGetView(
```
Do we need the translation from View to Window? It seems just okay to get Window directly from the registrar.

xiaowei-guan replied:
Good suggestion, for currently status, we can get window directly form the registrar. Thank you~

swift-kim commented on shell/platform/tizen/tizen_renderer.h:
```diff
@@ -29,6 +29,7 @@ class TizenRenderer {
 
   bool IsValid() { return is_valid_; }
 
+  virtual void* GetWindow() = 0;
```
I just realized that there's already `GetWindowHandle`.

xiaowei-guan replied:
yes, but EvasGL renderer return nullptr.
Hi,@bwikbs do we could  return evas_window_ by GetWindowHandle?

bwikbs replied:
@xiaowei-guan 
Sorry for late reply!
> Hi,@bwikbs do we could return evas_window_ by GetWindowHandle?

Of course it is possible! :smil

xiaowei-guan replied:
ok,Thank you~

xiaowei-guan committed:
Change interface name

1.Change interface name, make it more clearly.
2.Move HandleForRender and RendererFromHandle to unnamed namspace.

xiaowei-guan committed:
Get window from FlutterDesktopPluginRegistrarRef

bwikbs reviewed (approved):
I guess platform view should also use this!

xiaowei-guan committed:
Merge GetWindow and GetWindowHandle
